sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -1208,85 +1208,172 @@ class ColumnName(name: String) extends Column(name) {
    */
   def boolean: StructField = StructField(name, BooleanType)
 
+  /**
+   * Creates a new `StructField` of type boolean.
+   * @since 2.3.0
+   */
+  def boolean(nullable: Boolean): StructField = StructField(name, BooleanType, nullable)
+
   /**
    * Creates a new `StructField` of type byte.
    * @since 1.3.0
    */
   def byte: StructField = StructField(name, ByteType)
 
+  /**
+   * Creates a new `StructField` of type byte.
+   * @since 2.3.0
```

Comment:
`2.4.0` and in the other places too (unless they patch 2.3.0 and becomes 2.3.1).